quilt3distribute/dataset.py

```diff
@@ -183,6 +189,30 @@ def set_extra_files(self, files: Union[List[Union[str, Path]], Dict[str, List[Un
         # Set the paths
         self.extra_files = converted
 
+    @staticmethod
+    def _recursive_clean(pkg: quilt3.Package, metadata_reduction_map: Dict[str, bool]):
+        # For all keys in current package level
+        for k in pkg:
+            # If it is a PackageEntry object, we know we have hit a leaf node
+            if isinstance(pkg[k], quilt3.packages.PackageEntry):
```

Comment:
Yes. You cannot put arbitrary objects into quilt3 packages like you used to be able to in quilt2. Because of this, we know that all objects follow the `PackageEntry` object definition.